### PR TITLE
Add terraform config

### PR DESCRIPTION
**Pull request by vipin-dfe**
RSM is being migrated to AKS. The first step is to be able to deploy review apps to validate the build and the terraform code. This PR deploys the RSM application to the AKS test cluster so it can be tested by developers. 
can be tested via https://refer-serious-misconduct-pr-1234.test.teacherservices.cloud/

### CODE REVIEW

**Comment by neillturner** [comment]
[RSM: Deploy review app using make](https://trello.com/c/RICWTsAA/2077-rsm-deploy-review-app-using-make)

File: config/environments/production.rb [hunk 1]
```diff
@@ -41,7 +41,7 @@
 
   # Force all access to the app over SSL, use Strict-Transport-Security, and use secure cookies.
   # config.force_ssl = true
-  config.force_ssl = true
+  config.force_ssl = false
```
Comment (saliceti) [inline config/environments/production.rb:44]: This should still be true
Can we please add an exception such as:
https://github.com/DFE-Digital/access-your-teaching-qualifications/blob/main/config/environments/production.rb#L46

File: terraform/application/application.tf [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  source = "./vendor/modules/aks//aks/application_configuration"
+
+  namespace              = var.namespace
+  environment            = var.environment
+  azure_resource_prefix  = var.azure_resource_prefix
+  service_short          = var.service_short
+  config_short           = var.config_short
+  secret_key_vault_short = "app"
+
+  # Delete for non rails apps
```
Comment (saliceti) [inline terraform/application/application.tf:11]: Can the comment be deleted?

File: terraform/application/application.tf [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  is_rails_application = true
+
+  config_variables = {
+    ENVIRONMENT_NAME           = var.environment
+    PGSSLMODE                  = local.postgres_ssl_mode
+    HOSTING_DOMAIN             = "refer-serious-misconduct-pr-1234.test.teacherservices.cloud" # var.domain != null ? "https://${var.domain}" : "https://${local.rsm_web_app_name}.azurewebsites.net"
```
Comment (saliceti) [inline terraform/application/application.tf:17]: Can we build this in a variable instead of hardcoding the value?

Comment (saliceti) [reply terraform/application/application.tf:17]: Like: https://github.com/DFE-Digital/access-your-teaching-qualifications/blob/main/terraform/application/variables.tf#L100

File: terraform/application/application.tf [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    HOSTING_ENVIRONMENT_NAME   = var.environment                                               # local.hosting_environment
```
Comment (saliceti) [inline terraform/application/application.tf:18]: Can the comment be deleted?

File: terraform/application/application.tf [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    RAILS_SERVE_STATIC_FILES   = "true"
```
Comment (saliceti) [inline terraform/application/application.tf:19]: This is not needed as it comes with the option `is_rails_application = true`. See: https://github.com/DFE-Digital/terraform-modules/blob/main/aks/application_configuration/resources.tf#L2C3-L11

File: terraform/application/application.tf [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+  is_web                     = false
+  docker_image               = var.docker_image
+  replicas                   = var.worker_replicas
+  max_memory                 = var.worker_memory_max
+  cluster_configuration_map  = module.cluster_data.configuration_map
+  kubernetes_config_map_name = module.application_configuration.kubernetes_config_map_name
+  kubernetes_secret_name     = module.application_configuration.kubernetes_secret_name
+  command                    = ["/bin/sh", "-c", "bundle exec sidekiq -C config/sidekiq.yml"]
+  probe_command              = ["pgrep", "-f", "sidekiq"]
+  enable_gcp_wif             = true
```
Comment (saliceti) [inline terraform/application/application.tf:68]: Can this be removed for now? gcp_wif is not enabled yet

File: terraform/application/config/development.tfvars.json
Comment (saliceti) [inline terraform/application/config/development.tfvars.json:1]: Can we remove the file so we focus on the review app?

File: terraform/application/config/development_Terrafile
Comment (saliceti) [inline terraform/application/config/development_Terrafile:1]: Can we remove the file so we focus on the review app?

File: terraform/application/config/production.tfvars.json
Comment (saliceti) [inline terraform/application/config/production.tfvars.json:1]: Can we remove the file so we focus on the review app?

File: terraform/application/config/production_Terrafile
Comment (saliceti) [inline terraform/application/config/production_Terrafile:1]: Can we remove the file so we focus on the review app?

File: terraform/domains/environment_domains/config/development.tfvars.json
Comment (saliceti) [inline terraform/domains/environment_domains/config/development.tfvars.json:1]: Can we remove all the domain files so we focus on the review app?

File: terraform/application/application.tf (hunk 3 above)
Comment (vipin-dfe) [reply terraform/application/application.tf:17]: done.

File: terraform/application/application.tf (hunk 2 above)
Comment (vipin-dfe) [reply terraform/application/application.tf:11]: done.

File: terraform/application/application.tf (hunk 6 above)
Comment (vipin-dfe) [reply terraform/application/application.tf:68]: done.

File: terraform/application/config/development_Terrafile
Comment (vipin-dfe) [reply terraform/application/config/development_Terrafile:1]: done.

File: terraform/domains/environment_domains/config/development.tfvars.json
Comment (vipin-dfe) [reply terraform/domains/environment_domains/config/development.tfvars.json:1]: done.

File: terraform/application/config/production_Terrafile
Comment (vipin-dfe) [reply terraform/application/config/production_Terrafile:1]: done.

File: terraform/application/config/production.tfvars.json
Comment (vipin-dfe) [reply terraform/application/config/production.tfvars.json:1]: done.

File: terraform/application/config/development.tfvars.json
Comment (vipin-dfe) [reply terraform/application/config/development.tfvars.json:1]: done.

File: config/environments/production.rb (hunk 1 above)
Comment (vipin-dfe) [reply config/environments/production.rb]: done.

File: terraform/application/application.tf (hunk 4 above)
Comment (vipin-dfe) [reply terraform/application/application.tf]: done.

File: terraform/application/application.tf (hunk 5 above)
Comment (vipin-dfe) [reply terraform/application/application.tf]: done.

File: terraform/application/variables.tf [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+  default = 1
+}
+
+locals {
+  postgres_ssl_mode                = var.enable_postgres_ssl ? "require" : "disable"
+  storage_account_environment      = var.config == var.environment ? var.config_short : replace(var.environment, "-", "")
+  allegations_storage_account_name = "${var.azure_resource_prefix}rsmalleg${local.storage_account_environment}sa"
+
+  environment_variables = yamldecode(file("${path.module}/config/${var.config}.yml"))
+  access_domain          = "${var.service_name}-${var.environment}.${module.cluster_data.ingress_domain}"
```
Comment (saliceti) [inline terraform/application/variables.tf:83]: Could this be renamed to "ingress_domain"

File: terraform/application/variables.tf [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+  access_external_domain = try(local.environment_variables["ACCESS_EXTERNAL_DOMAIN"], local.access_domain)
```
Comment (saliceti) [inline terraform/application/variables.tf:84]: Could this be renamed to "external_domain"

**Review by saliceti — APPROVED** [review]

**Pull request merged**